design/open/lakefs-kv-execution-plan.md

```diff
@@ -51,22 +52,23 @@ When a complete data model / functionality completes the transition we can exclu
 
 ## Global changes
 
-- Transition from db to kb should include a configuration flag that enables the use of 'kv' vs the current 'db' implementation.
+- Transition from db to kv should include a configuration flag that enables the use of 'kv' vs the current 'db' implementation.
   As complete features move to use kv, and migration plan is implemented, we can remove the flag check for the feature and just enable the new functionality.
 - Migrate from current DDL to kv will be supported only for Postgres and dropped when all features works using kv.
-- Basic dump/restore functionality should be supported for the k/v implementation, how do we migrate the information - discovery
+- Dump and restore functionality implementation as part of ref-dump.
 
 ## Testing
 
-- Migration - running the same tests on data that was generated with previous db/model
-- Functionality performance - set of tests that will check that we didn't degrade
-- Migration from old format performance - can't take 24h to switch to new version (downtime?)
-- New locking mechanism functionality
+- Migration - upgrade from db to kv version and verify commit log, branch staging environment, authorization and actions data integrity.
+- Functionality - verify that all data kept on kv works with the current functionality (CRUD).
+- Ref-store dump/restore state test that old data is available and not corrupted.
+- Functionality performance - set of tests that will check that we didn't degrade.
+- Migration from old format performance - can't take 24h to switch to new version (downtime?).
+- New locking mechanism functionality. Multiple commits. Read and write while commit. Commit after a failed commit.
 
```

Comment:
I would add a comment that testing is a discovery item that should consider all the above aspects, and should be ready for the first milestone (i.e. we can't really start to migrate if we don't have proper tests).

Reply:
The discovery part is just to map how we're going to test the new functionality? Because test first and implementing everything without any new functionality can be hard. Some functionality should be backed with unit test.